### PR TITLE
8292695: SIGQUIT and jcmd attaching mechanism does not work with signal chaining library

### PR DESCRIPTION
**Pull request by GoeLin**
Resolved signals_posix.cpp because test for do_check is not in 17.
That was added by  8283337: Posix signal handler modification warning triggering incorrectly
I add 8283337 to this backport as it is subsumed by 8292695.

Resolved os.cpp due to context.
Context was changed by "8269466: Factor out the common code for initializing and starting internal VM JavaThreads"

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8292695](https://bugs.openjdk.org/browse/JDK-8292695): SIGQUIT and jcmd attaching mechanism does not work with signal chaining library
 * [JDK-8283337](https://bugs.openjdk.org/browse/JDK-8283337): Posix signal handler modification warning triggering incorrectly

### Reviewers
 * [Man Cao](https://openjdk.org/census#manc) (@caoman - no project role) ⚠️ Review applies to [67611936](https://git.openjdk.org/jdk17u-dev/pull/834/files/67611936b3c9f40508aa39226dc317664d7945d8)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev pull/834/head:pull/834` \
`$ git checkout pull/834`

Update a local copy of the PR: \
`$ git checkout pull/834` \
`$ git pull https://git.openjdk.org/jdk17u-dev pull/834/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 834`

View PR using the GUI difftool: \
`$ git pr show -t 834`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/834.diff">https://git.openjdk.org/jdk17u-dev/pull/834.diff</a>

</details>

### CODE REVIEW

**Comment by GoeLin** [comment]
Pre-submit test: timeout of javac

**Comment by caoman** [comment]
Thanks for the backport. Looks like https://bugs.openjdk.org/browse/JDK-8292695 didn't have "jdk17u-fix-request", and I just added the label.

The change to HotSpot looks good, but should we also include the `TestBreakSignalThreadDump.java` test in the backport?

**Comment by GoeLin** [comment]
Hi,
Thanks for looking at my change!
I removed the label again. It is not yet ready for labeling. The label should only be applied once the change is ready. Here the review is still missing.
You are right, I forgot the test.  I will add it.
Best regards,
  Goetz

**Review by caoman — APPROVED** [review]
Thanks for the update. LGTM now. Not an official Reviewer though.

File: src/hotspot/os/posix/signals_posix.cpp [hunk 1]
```diff
@@ -1246,7 +1241,8 @@ void set_signal_handler(int sig, bool do_check = true) {
 
   // Save handler setup for later checking
```
Comment (TheRealMDoerr) [inline src/hotspot/os/posix/signals_posix.cpp:1242]: JDK-8283337 changed this comment to: // Save handler setup for possible later checking

**Review by TheRealMDoerr — COMMENTED** [review]
Looks basically good. Only one comment change is missing.

**Review by TheRealMDoerr — APPROVED** [review]
Thumbs up! Thanks for backporting!

**Pull request closed without merging**